src/frontend/LayoutConverter.cpp

```diff
@@ -403,6 +403,22 @@ ConversionResult LayoutConverter::convertAvroLayout(QString path) {
   return saveLayout(layout, savePath);
 }
 
+ConversionResult LayoutConverter::saveLayout(QString path) {
+  QFile jsonFile(path);
+  if (!jsonFile.open(QIODevice::ReadOnly | QIODevice::Text)) {
+    return OpenError;
+  }
+  if (QJsonDocument::fromJson(jsonFile.readAll()).object()
+      .value("info").toObject()
+      .value("version").toString() != "2") {
+    return UnsupportedLayout;
```

Comment:
How about using the `LayoutConverter::convertLayoutFormat()` function here to perform the conversion if we detect an old layout version (version 1)?